mainhandler/vulnscan.go

```diff
@@ -94,127 +94,43 @@ func convertImagesToWebsocketScanCommand(registry *registryScan, sessionObj *uti
-	}
-
-	if err != nil {
-		// if configmap not found, it means we will use all images and default depth
-		if strings.Contains(err.Error(), fmt.Sprintf("reason: configmaps \"%v\" not found", registryScanConfigmap)) {
-			glog.Infof("configmap: %s does not exists, using default values", registryScanConfigmap)
-			return NewRegistryScanConfig(registryName), string(cmDefaultMode), nil
-		} else {
-			return nil, string(cmDefaultMode), err
-		}
-	}
-	configData := configMap.GetData()
-	var registriesConfigs []registryScanConfig
-	registriesConfigStr, ok := configData["registries"].(string)
-	if !ok {
-		return nil, string(cmDefaultMode), fmt.Errorf("error parsing %v confgimap: registries field not found", registryScanConfigmap)
+func (actionHandler *ActionHandler) loadRegistryScan(sessionObj *utils.SessionObj) (*registryScan, error) {
+	registryScan := NewRegistryScan(actionHandler.k8sAPI)
+	var err error
+	if registryScan.isParseRegistryFromCommand(sessionObj.Command) {
+		err = registryScan.parseRegistryFromCommand(sessionObj)
+	} else {
+		err = registryScan.parseRegistryFromCluster(sessionObj)
 	}
-	registriesConfigStr = strings.Replace(registriesConfigStr, "\n", "", -1)
-	err = json.Unmarshal([]byte(registriesConfigStr), &registriesConfigs)
 	if err != nil {
-		return nil, string(cmDefaultMode), fmt.Errorf("error parsing ConfigMap: %s", err.Error())
-	}
-	for _, config := range registriesConfigs {
-		if config.Registry == registryName {
-			return &config, string(cmLoadedMode), nil
-		}
-	}
-	return NewRegistryScanConfig(registryName), string(cmDefaultMode), nil
-
-}
-
-func (actionHandler *ActionHandler) getRegistryScanSecret() (k8sinterface.IWorkload, error) {
-	secret, err := actionHandler.k8sAPI.GetWorkload(armotypes.KubescapeNamespace, "Secret", armotypes.RegistryScanSecretName)
-	if err == nil && secret != nil {
-		return secret, err
+		return nil, err
 	}
-
-	// deprecated namespace
-	secret, err = actionHandler.k8sAPI.GetWorkload(armotypes.ArmoSystemNamespace, "Secret", armotypes.RegistryScanSecretName)
-	return secret, err
-
+	return &registryScan, nil
 }
 
-func (actionHandler *ActionHandler) scanRegistries(sessionObj *utils.SessionObj) error {
-
-	/*
-		Auth data must be stored in kubescape-registry-scan secret
-		Config data must be stored in "kubescape-registry-scan" config map
-	*/
-
-	registryName, err := actionHandler.parseRegistryNameArg(sessionObj)
+func (actionHandler *ActionHandler) testRegistryConnectivity(sessionObj *utils.SessionObj) error {
+	registryScan, err := actionHandler.loadRegistryScan(sessionObj)
 	if err != nil {
-		return fmt.Errorf("parseRegistryNameArg failed with err %v", err)
-	}
-	auth, err := actionHandler.getRegistryAuth(registryName)
-	if err != nil {
-		return fmt.Errorf("get registry auth failed with err %v", err)
+		glog.Errorf("in testRegistryConnectivity: loadRegistryScan failed with error: %v", err.Error())
```

Comment:
should we update the job report details

Reply:
This is error in command so I didn't use the new statuses on it at all. The error will be sent as error in the job report. Should I update details which the step failed as well?

Reply:
Yes